modules/angular-meteor-meteorCollection.js

```diff
@@ -142,15 +132,28 @@ angularMeteorCollections.factory('AngularMeteorCollection', ['$q', '$meteorSubsc
     AngularMeteorCollection.updateCursor = function (cursor) {
       var self = this;
 
+      var promise;
+      // Function applies async to combine multiple operations (savings, deletions etc)
+      // in one processing.
       function safeApply() {
+        if (promise) {
+          $timeout.cancel(promise);
+          promise = null;
+        }
         // Clearing the watch is needed so no updates are sent to server
-        // while handling updates from the server
-        self.UPDATING_FROM_SERVER = true;
-        if (!$rootScope.$$phase) $rootScope.$apply();
-        // Making sure we are setting to false only after one digest cycle and not before
-        $timeout(function(){
+        // while handling updates from the server.
+        if (!self.UPDATING_FROM_SERVER) {
+          self.UPDATING_FROM_SERVER = true;
+          $rootScope.$apply();
+        }
+        promise = $timeout(function () {
+          // Saves changes only when new docs have been added.
+          if (self.length != self._serverBackup.length) {
```

Comment:
I removed the line ````self.length != self._serverBackup.length````
Now all changes happened (incl. just updates) to the objects on the client should be handled in the updateCollection in the $timeout.